### PR TITLE
Add a hook for processing AuthenticatedUser

### PR DESCRIPTION
**Pull request by myui**
We would like to store WF invoked user information in TD.  So, add hook.

Example usage:

```java
    @Overrided
    protected void putAdditionalParams(@Nonnull Config params)
    {
        AuthenticatedUser user = getAuthenticatedUser();
        final Config info = user.getUserInfo().getNested("td").getNested("user");
        if (info.has("id")) {
            params.set("user_id", info.get("id", Integer.class));
        }
        if (info.has("email")) {
            params.set("user_email", info.get("email", String.class));
        }

        final Config context = user.getUserContext().getNested("td").getNested("user");
        if (context.has("ip_address")) {
            params.set("ip_address", context.get("ip_address", String.class));
        } else if (context.has("ip")) {
            params.set("ip_address", context.get("ip", String.class));
        }
    }
```

### CODE REVIEW

File: digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java [hunk 1]
```diff
@@ -256,10 +258,13 @@ public Response startAttempt(RestSessionAttemptRequest request)
                     .transform(r -> collectResumingTasks(r))
                     .or(ImmutableList.of());
 
+            Config params = request.getParams();
+            putAdditionalParams(params);
```
Comment (yoyama) [inline digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java:262]: IIUC `request.params` store user defined parameters.
So mix of system and user defined may be harmful.

Comment (yoyama) [reply digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java:262]: Ah scheduler may write some parameters to it. Let me check it.

Comment (yoyama) [reply digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java:262]: Ok, I confirmed both user define and system defined params are stored.

**Review by yoyama — APPROVED** [review]

File: digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java (hunk 1 above)
Comment (myui) [reply digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java:262]: so, then `params.set("__td__.user_id", info.get("id", Integer.class));` or some.

**Pull request merged**